core/src/main/java/com/exadel/etoolbox/querykit/core/models/query/QueryStatisticModel.java

```diff
@@ -0,0 +1,256 @@
+
+        for(int i = 0; i < queries.size(); i++) {
+            JsonObject jsonObject = queries.get(i).getAsJsonObject();
+            String statement = getAsString(jsonObject, "query");
+            if (StringUtils.contains(statement, "oak-internal") || StringUtils.startsWithIgnoreCase(statement, "explain ")) {
+                continue;
+            }
+            String language = getAsString(jsonObject, "language");
+            String thread = getAsString(jsonObject, "lastThreadName");
+            String lastExecuted = getAsString(jsonObject, "lastExecutedMillis");
+            int executeCount = getAsInt(jsonObject, "executeCount");
+
+            QueryStatisticModel.QueryInfo queryInfo = new QueryStatisticModel.QueryInfo(queryManager);
+            queryInfo.setLanguage(language);
+            queryInfo.setStatement(statement);
+            queryInfo.setTheadName(thread);
+            queryInfo.setExecuteCount(executeCount);
+            queryInfo.setLastExecuted(lastExecuted);
+            result.add(queryInfo);
+        }
+        result.sort((a, b) -> Integer.compare(a.getExecuteCount(), b.getExecuteCount()) * -1);
+        return result;
+    }
+
+    private ObjectName getQueryStatMBean(MBeanServerConnection server) {
+        try {
+            Set<ObjectName> names = server.queryNames(new ObjectName("org.apache.jackrabbit.oak:type=QueryStats,*"), null);
+            return names.iterator().next();
+        } catch (IOException | MalformedObjectNameException | NoSuchElementException e) {
+            return null;
+        }
+    }
+
+    private List<String> getIndexes(ResourceResolver resourceResolver) {
+        Resource oakIndex = resourceResolver.getResource(OAK_INDEX_PREFIX);
+        if (oakIndex == null || !oakIndex.hasChildren()) {
+            return Collections.emptyList();
+        }
+        return StreamSupport.stream(oakIndex.getChildren().spliterator(), false)
+                .map(Resource::getPath)
+                .sorted()
+                .collect(Collectors.toList());
+    }
+
+    private String getIndexId(String value) {
+        if (value.contains("(") && value.contains(")")) {
+            return StringUtils.substringBefore(value, "(").trim();
+        } else if (StringUtils.containsIgnoreCase(value, OAK_INDEX_PREFIX)) {
+            return StringUtils.strip(StringUtils.substringAfter(value, OAK_INDEX_PREFIX), STRIP_CHARS);
+        }
+        return StringUtils.strip(value, STRIP_CHARS);
+    }
+
+    private String getAsString(JsonObject object, String property) {
+        try {
+            return object.get(property).getAsString();
+        } catch (UnsupportedOperationException e) {
+            return StringUtils.EMPTY;
+        }
+    }
+
+    private int getAsInt(JsonObject object, String property) {
+        try {
+            return object.get(property).getAsInt();
+        } catch (UnsupportedOperationException e) {
+            return 0;
+        }
+    }
+
+    public static class QueryInfo {
```

Comment:
1) I'd rather extracted into a separate class file
2) Please refactor to use Lombok. It's Lombok throughout this project

Reply:
Done